pennylane/_qubit_device.py

```diff
@@ -1219,6 +1231,12 @@ def classical_shadow(self, wires, n_snapshots, circuit, seed=None):
 
         return self._cast(self._stack([outcomes, recipes]), dtype=np.int8)
 
+    def classical_shadow_expval(self, obs, circuit):
+        """TODO: docs"""
+        bits, recipes = self.classical_shadow(obs, circuit)
+        shadow = ClassicalShadow(bits, recipes, wire_map=obs.wires.tolist())
+        return shadow.expval(obs.H, obs.k)
```

Comment:
I was thinking of adding a flag `return_bits` that then additionally returns the `bits` and `recipes`. Do you see any potential problem from this?

Reply:
Couldnt we conform to the new return types and explicitly return a tuple `(expval, bits, recipes)`?

Reply:
I have limited context here, but two things come to mind:
1. If the output is something that's expected to be a ragged array, it could be handled by the original pipeline so long as there are no errors. Ragged arrays emit deprecation warnings. Having said that, the output of `qml.counts()` is for example an exception across the code base and is not cast using an array. The situation there is somewhat different though: dictionaries cannot be cast to tensor-like objects with some frameworks.
2. One approach is to have a `tuple` output specifically for the new return types that are already experimentally available. There could be two parts of the logic, one performing operations `if qml.active_return()`, in which case the new return types have been turned on.

Reply:
I think we can leave this for now, and think about adding an option to return the `bits` and `recipes` as an enhancement later on.